sdk/src/wallet/core/operations/stronghold_backup/mod.rs

```diff
@@ -79,42 +81,40 @@ impl Wallet {
             return Err(crate::wallet::Error::Backup("backup path doesn't exist"));
         }
 
-        let wallet_data = self.data().await;
+        let wallet_ledger = self.ledger().await;
 
         // We don't want to overwrite a possible existing wallet
-        if !wallet_data.outputs.is_empty() {
+        if !wallet_ledger.outputs.is_empty() {
             return Err(crate::wallet::Error::Backup(
                 "can't restore backup when there is already a wallet",
             ));
         }
 
-        let curr_bip_path = wallet_data.bip_path;
+        let curr_bip_path = self.bip_path().await;
 
         // Explicitly drop the data to avoid contention
-        drop(wallet_data);
+        drop(wallet_ledger);
 
         // We'll create a new stronghold to load the backup
         let new_stronghold = StrongholdSecretManager::builder()
             .password(stronghold_password.clone())
             .build(backup_path.clone())?;
 
-        let (read_client_options, read_secret_manager, read_wallet_data) =
-            read_wallet_data_from_stronghold_snapshot::<SecretManager>(&new_stronghold).await?;
-
-        let read_bip_path = read_wallet_data.as_ref().and_then(|data| data.bip_path);
+        let (read_address, read_bip_path, read_alias, read_client_options, read_secret_manager, read_wallet_ledger) =
+            restore_from_stronghold_snapshot::<SecretManager>(&new_stronghold).await?;
 
         // If the bip path is not matching the current one, we may ignore the backup
         let ignore_backup_values = ignore_if_bip_path_mismatch.map_or(false, |ignore| {
             if ignore {
-                // TODO: #1279 okay that if both are none we always load the backup values?
+                // TODO: is it okay that if both are none we always load the backup values?
```

Comment:
Do we have an issue for this?

Reply:
created an issue: https://github.com/iotaledger/iota-sdk/issues/2032